### ISSUE TITLE
Viva piñata

### ISSUE THREAD

**Issue by minuteworld1** [ValveSoftware/Proton#6060]
# Compatibility Report
- Name of the game with compatibility issues: viva piñata 
- Steam AppID of the game: Non-steam game

## System Information
- GPU: steamdeck
- Driver/LLVM version:  steamdeck
- Kernel version: steamdeck
- Link to full system information report as [Gist](https://gist.github.com/):steamdeck
- Proton version: 3.70

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 
there's a native pc port of viva pinata and ive been trying to get this to run for hours. got the .dll file that strips away GFWL and everything. works fine on my windows 10 pc with it showing the launcher and you can launch the game fine but on steam OS beit steam's proton or ge proton, it just takes me to the launcher and the launcher is broken with it being a blank static image with no way to click "play". I'm assuming this is a proton problem and i should just wait?

## Reproduction
go to old games download and download the game, install it on a windows machine, transfer it to steamdeck, get the gfwl removal patch here https://community.pcgamingwiki.com/files/file/576-bulletstorm-xlivedll/, paste it in game folder, add  "startup.exe" as a non steam game, use any version of proton, click on "yes" and it'll take you to the static launcher with no words.

**Comment by kisak-valve** [comment]
Hello @minuteworld1, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Since this is a non-steam game, $APPID will most likely be a large random number. Also, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by minuteworld1** [comment]
right. i'll try to do all that when i get home but now ill just share a screenshot
![20220805083121_1](https://user-images.githubusercontent.com/77810087/183080292-91f8d2eb-2b48-4d92-894a-ff965ab8a0f8.jpg)

**Comment by minuteworld1** [comment]
alright so i added the launch command. where exactly do i find the .log and how do i do the gist again? i'm sorry, i am very new to linux and i am not literate in this OS, only windows. I will try to share the log and gist when i get home from work. 

if you guys could get this game running eventually, i'd be extremely happy.

**Comment by kisak-valve** [comment]
Proton logs should end up in the home folder by default, which is accessible on the Steam Deck using the desktop mode, the same with the system information from the Steam client running in the desktop session.

**Comment by minuteworld1** [comment]
gist:

https://gist.github.com/minuteworld1/df649f1f3091a4d8e8e9d4286490d747

**Comment by minuteworld1** [comment]
log:  
[steam-16822930425046368256.log](https://github.com/ValveSoftware/Proton/files/9274400/steam-16822930425046368256.log)

**Comment by minuteworld1** [comment]
hope i did that right

**Comment by minuteworld1** [comment]
no rush but i'd really like this game too work. launcher still looks like that in 7.0-4 and experimental and ge. i literally think if i can get past the launcher, the game will work fine

**Comment by minuteworld1** [comment]
I get that static image in lutris as well.

**Comment by RedRover24** [comment]
So i've managed to find the fix for getting past the launcher by editing the startup.xml.

Startup.xml is found in /home/deck/.steam/steam/steamapps/compatdata/(GAMEID)/pfx/drive_c/users/steamuser/AppData/Roaming/Microsoft Games/Viva Pinata

Set false to true for "DontShow" and "DontShowUntilNew"

Only problem now is that there are major graphical issues and lack of any audio, rendering it completely unplayable as of writing this. Tested both on Proton experimental and 7.0-4.

**Comment by grantbevis** [comment]
@minuteworld1 I'm having mild success on the deck with this guide. I can get the game to run but I'm unable to complete the patch to make the game widescreen

https://reddit.com/r/VivaPinata/comments/jke4er/viva_pinata_gnulinux_installation_guide/

**Comment by JourDANGER** [comment]
I was also able to get the launcher to run this way. Without text or any way to interact with the app or start the game. 
 
> right. i'll try to do all that when i get home but now ill just share a screenshot ![20220805083121_1](https://user-images.githubusercontent.com/77810087/183080292-91f8d2eb-2b48-4d92-894a-ff965ab8a0f8.jpg)

I found a way to bypass the launcher and get the game to run consistently.

To get the skip launcher you need to edit a Startup.xml file in the numbered compatdata subfolder for the game. You may need to run the startup.exe at least once to generate the Startup.xml in the compatdata subfolder for the game. You can find the compatdata subfolder number using the proton tricks app. 

In my case this was 4181416414. Yours will be different. My compatdata folder containing the numbered subfolders was located here `/home/deck/.local/share/Steam/steamapps/compatdata`. Under the numbered subfolder for the game, in my case 4181416414. Find the Startup.xml file. The default location of this file should be `/pfx/drive_c/users/steamuser/AppData/Roaming/Microsoft Games/Viva Pinata/Startup.xml`

The full path to my file was 
`/home/deck/.local/share/Steam/steamapps/compatdata/4181416414/pfx/drive_c/users/steamuser/AppData/Roaming/Microsoft Games/Viva Pinata/Startup.xml`

Edit the file to match the this
`<Startup DontShow="true" DontShowUntilNew="true" ContentVersion="1"/>`

The game launches consistently and runs great.

![20240810225410_1](https://github.com/user-attachments/assets/d30c97c0-8900-48ee-a1aa-7b71b3f84cfb)

**Comment by 27hectormanuel** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6060#issuecomment-2282636861

Mine has this weird bug where it has a black screen and flickers black when the camera moves a certain way
